datalad/metadata/search.py

```diff
@@ -124,6 +124,8 @@ def __call__(match,
                   match, dataset)
         try:
             ds = require_dataset(dataset, check_installed=True, purpose='dataset search')
+            if ds.id is None:
+                raise NoDatasetArgumentFound
```

Comment:
I will pickup 107bb8b6e33d1775f872275fc302878dcae43434 from https://github.com/datalad/datalad/pull/1010 which is WiP